### PR TITLE
mcux: Build fsl_elcdif if CONFIG_VIDEO_MCUX_ELCDIF

### PR DESCRIPTION
**Pull request by loicpoulain**
Add fsl_elcdif if the zephyr video mcux driver is enabled.
We keep the *DISPLAY* flag for now for smoother integration.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

### CODE REVIEW

**Comment by loicpoulain** [comment]
@MaureenHelm could you review this one

File: mcux/drivers/imx/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ zephyr_library_compile_definitions_ifdef(
 
 zephyr_sources_ifdef(CONFIG_HAS_MCUX_CACHE    fsl_cache.c)
 zephyr_sources_ifdef(CONFIG_DISPLAY_MCUX_ELCDIF	fsl_elcdif.c)
+zephyr_sources_ifdef(CONFIG_VIDEO_MCUX_ELCDIF  fsl_elcdif.c)
```
Comment (galak) [inline mcux/drivers/imx/CMakeLists.txt]: why do we need a different Kconfig symbol, cant you use CONFIG_DISPLAY_MCUX_ELCDIF?

Comment (loicpoulain) [reply mcux/drivers/imx/CMakeLists.txt]: We don't need different symbols, actually after #21009 merge [1], we should not need the CONFIG_DISPLAY_MCUX_ELCDIF anymore here. The point is that display_mcux_elcdif driver will be moved to video_mcux_elcdif driver.

[1] https://github.com/zephyrproject-rtos/zephyr/pull/21009

**Comment by dleach02** [comment]
@loicpoulain, I'm closing this PR as part of a clean up pass. If you are still interested in enabling this drive please resubmit to align with the current HAL structure.

**Pull request closed without merging**